libs/langchain/pyproject.toml

```diff
@@ -119,7 +120,7 @@ cassio = "^0.1.0"
 langchainhub = "^0.1.16"
 
 [tool.poetry.group.lint.dependencies]
-ruff = "^0.5"
+ruff = "~0.9.1"
```

Comment:
let's keep this as `^0.9` to avoid auto updating and pulling in breaking changes

Reply:
Breaking changes can happen between minor versions in ruff (new rules, different formatting).
`^0.9` is equivalent to `>=0.9, <1.0.0` whereas `~0.9.1` is equivalent to `>=0.9, <0.10.0`. So I think the latter is better.

Reply:
TIL they're actually equivalent in this case https://python-poetry.org/docs/dependency-specification/#caret-requirements

we tend to use `^` across other dependencies so will keep there

Reply:
Oh yes for version <1, caret rules are different ! (TIL too 😄)